doc/enums.asciidoc

```diff
@@ -0,0 +1,155 @@
+== Enums ==
+
+Golo allows the definition of sum algebraic data types, also known as
+_tagged_union_
+footnote:[ http://en.wikipedia.org/wiki/Tagged_union[Wikipedia: Tagged union]
+], as present in many functional languages: OCaml, Haskell, Rust, Scala to
+name a few.
+The dual algebraic data type, the product type is provided by `struct` and `tuple`.
+
+=== Definition ===
+
+Unions are defined at the module-level:
+[source,golo]
+----
+module sample
+
+enum Option = {
+ Some = { value }
+ None
+}
+
+function main = |args| {
+  let aString = Option.Some("Hello")
+  println(aString: value())
+
+  let noString = Option.None()
+  println(noString)
+}
+----
+
+=== Usage example ===
+
+Some well known usages of sum types are:
+
+==== Enumerations
```

Comment:
I think you can go for a consistent header style rather than sometimes `== foo ==` and sometimes `== foo` :smiley: